### DISCUSSION TITLE
How do I use the service_key in an API call? (Nuxt 3)

### DISCUSSION THREAD

**Discussion by phillipmohr** [supabase/supabase#9101] [Questions] [answered]
Hello, I use Nuxt 3 RC 11 together with `"@nuxtjs/supabase": "^0.1.25",`

I would like to list all users and I've read that I can do that with the help of the service_role key, but where do I need to include the key?

`/server/api/users.ts`
```ts
import { serverSupabaseClient } from '#supabase/server'

export default defineEventHandler(async ({ event, req }) => {
    const client = serverSupabaseClient(event)

    const { data: users, error } = await client.from('users').select('*')

    return users
})
```

I get: `permission denied for view users` with code `42501`

**Comment by GaryAustin1** [comment]
There is an auth API call to list users that uses the service_role (but I think it is limited to returning the first 50 or something like that).  https://supabase.com/docs/reference/javascript/auth-api-listusers

You can not access the auth.users table (or the auth schema) thru the api directly.  If that is what 'users' above refers to versus a public table copy of user info.  If you have public profile/user table then you can access that with or without the service_role key depending on your RLS.

Otherwise you can do an rpc call with "security definer" declaration and access auth.users with sql and return what you would like.

Service_role key is used in client init where you put the anon key.

**Comment by phillipmohr** [comment, marked as answer]
@GaryAustin1 Thank you. the listUsers function works fine for now.

I got it working like this

```ts
import { serverSupabaseServiceRole } from '#supabase/server'

export default defineEventHandler(async ({ event, req }) => {
    const supabase = serverSupabaseServiceRole(event)
    const { data: users, error } = await supabase.auth.api.listUsers()

    if (error) {
        return error
    }

    return users
})
```
